lib/method/registerMethodFactory.js

```diff
@@ -0,0 +1,142 @@
+const entropy = require('@dashevo/dpp/lib/util/entropy');
+
+const createDocumentTransactionData = require(
+  '../transaction/createDocumentTransactionData',
+);
+
+const { hash } = require('../utils/doubleSha256Multihash');
+const waitWithAttempts = require('../utils/waitWithAttempts');
+
+/**
+ * Register a domain (factory)
+ *
+ * @param {DAPIClient} dapiClient
+ * @param {DashPlatformProtocol} dpp
+ * @param {wallet} wallet
+ * @param {buser} bUser
+ *
+ * @return {registerMethod}
+ */
+function registerMethodFactory(dapiClient, dpp, wallet, bUser) {
+  /**
+   * Register a domain
+   *
+   * @typedef registerMethod
+   *
+   * @param {string} name
+   *
+   * @returns {Promise<Document>}
+   */
+  async function registerMethod(name) {
+    dpp.setUserId(bUser.regTxId);
+
+    const account = wallet.getAccount();
+    const { address } = account.getAddress();
+    const [privateKey] = account.getPrivateKeys([address]);
```

Comment:
It doesn't look like a user-based derived key. There was a DIP about purpose-based key derivation, but I'm not sure that the wallet lib supports that. I guess this would be just a random private key otherwise

Reply:
My point is: there's a specific key used to manage user. We haven't decided how to derive that key though. My suggestion would be to add `.getIdentityPrivateKey(index: number)` to the wallet-lib, and use that method

Reply:
I think we could postpone this part since we don’t have identities and wallet lib doesn’t support it.

Reply:
Let's add the task to Jira then